src/components/WeatherByHourData.js

```diff
@@ -0,0 +1,84 @@
+import axios from 'axios'
+import React, { useEffect, useState } from 'react'
+
+const WeatherByHourData = (results) => {
+    const [weatherData, setWeatherData] = useState([]);
+    const [count, setCount] = useState(0)
+    const timeOption = results.timeOption;
+    const date = results.date.toISOString().slice(0, 10);
+    const cityName = results.results.name;
+
+    useEffect(() => {
+        axios.get(`https://api.openweathermap.org/data/2.5/forecast?q=${cityName}&units=metric&appid=${process.env.REACT_APP_APIKEY}`)
+            .then(res => {
+                setWeatherData(res.data.list)
+                setCount(count + 1)
+            })
+            .catch(err => {
+                console.log(err.message)
+            })
+
+
+    }, [cityName, date, timeOption])
+
+    useEffect(() => {
+        checkChildForErrorMessage()
+    }, [])
+
+
+    const filterDataByDate = (date, timeOption) => {
+
+        if (timeOption === "SelectTime" || timeOption === '') {
+            return <div className='error-message'>
+                <h1>Please select a time and date to get weather information</h1>
```

Comment:
UX suggestion here. The error occurs when a time isn't entered. I feel it would be better to indicate just that in the error message. Something like "Please select a time" instead of "time and date".